kuzzleio/kuzzle
# Support script_fields query

## What does this PR do ?

Allow support of Elasticsearch [script fields](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/search-request-body.html#request-body-search-script-fields) in search body.

Fix https://github.com/kuzzleio/kuzzle/issues/1541

### How should this be manually tested?

Use this script, you should see the `fields` property on each hit:
```
const {
  Kuzzle,
  WebSocket
} = require('kuzzle-sdk')

const kuzzle = new Kuzzle(new WebSocket('localhost'))

kuzzle.on('networkError', err => console.log(err));

(async () => {
  try {
    await kuzzle.connect();
    const query = {
      "script_fields": {
        "ageValue": {
          "script": {
            "lang": "painless",
            "source": "doc['age'].value"
          }
        }
      },
      "query": {}
    };

    if (! await kuzzle.collection.exists('index', 'collection')) {
      await kuzzle.index.create('index');
      await kuzzle.collection.create('index', 'collection');
      await kuzzle.document.create('index', 'collection', { age: 21 });
      await kuzzle.document.create('index', 'collection', { age: 42 });
      await kuzzle.collection.refresh('index', 'collection');
    }

    const res = await kuzzle.document.search('index', 'collection', query);
    console.log(JSON.stringify(res.hits, null, 2));
  }
  catch (error) {
    console.log(error);
  }
  finally {
    kuzzle.disconnect();
  }
})();
```


### Boyscout

 - async/await and Array.map
